src/shogun/clustering/GMM.cpp

```diff
@@ -185,6 +185,10 @@ float64_t CGMM::train_em(float64_t min_cov, int32_t max_iter, float64_t min_chan
 		pb.print_progress();
 		max_likelihood(alpha, min_cov);
 
+		this->observe<float64_t>(iter, "log_likelihood", "Log Likelihood", log_likelihood_cur);
+		this->observe<SGVector<float64_t>>(iter, "coefficients", "Mixture Coefficients", alpha);
+		this->observe<std::vector<CGaussian*>>(iter, "components");
```

Comment:
and these would have to be cloned actually (unless they are just printed)

Reply:
Same here. They are all copied/cloned :)